js/ui/notificationDaemon.js

```diff
@@ -713,6 +713,10 @@ var GtkNotificationDaemonAppSource = new Lang.Class({
         return new FdoApplicationProxy(Gio.DBus.session, this._appId, this._objectPath, callback);
     },
 
+    _createNotification: function(params) {
+        return new GtkNotificationDaemonNotification(this, params);
+    },
+
```

Comment:
Yeah. I tried to avoid this, so why I when for the prototype solution. I guess it's fine, especially since you sent it upstream too.